pkg/index/index_factory_test.go

```diff
@@ -0,0 +1,199 @@
+		h.AssertNil(t, err)
+		indexFactory = index.NewIndexFactory(authn.DefaultKeychain, tmpDir)
+	})
+
+	it.After(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	when("#CreateIndex", func() {
+		it.Before(func() {
+			indexRepoName = h.NewRandomIndexRepoName()
+		})
+
+		when("no options are provided", func() {
+			it("creates an image index", func() {
+				imageIndex, err = indexFactory.CreateIndex(indexRepoName)
+				h.AssertNil(t, err)
+				h.AssertNotNil(t, imageIndex)
+			})
+		})
+	})
+
+	when("#Exists", func() {
+		when("index exists on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				setUpLocalIndex(t, indexFactory, indexRepoName)
+			})
+
+			it("returns true", func() {
+				h.AssertTrue(t, indexFactory.Exists(indexRepoName))
+			})
+		})
+
+		when("index does not exist on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+			})
+
+			it("returns false", func() {
+				h.AssertFalse(t, indexFactory.Exists(indexRepoName))
+			})
+		})
+	})
+
+	when("#LoadIndex", func() {
+		when("index exists on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				setUpLocalIndex(t, indexFactory, indexRepoName)
+			})
+
+			it("loads the index from disk", func() {
+				imageIndex, err = indexFactory.LoadIndex(indexRepoName)
+				h.AssertNil(t, err)
+				h.AssertNotNil(t, imageIndex)
+			})
+		})
+
+		when("index does not exist on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+			})
+
+			it("errors a message", func() {
+				_, err = indexFactory.LoadIndex(indexRepoName)
+				h.AssertError(t, err, fmt.Sprintf("Image: '%s' not found", indexRepoName))
+			})
+		})
+	})
+
+	when("#FetchIndex", func() {
+		when("index exists in a remote registry", func() {
+			var remoteIndexRepoName string
+
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				remoteIndexRepoName = newTestImageIndexName("fetch-remote")
+				setUpRandomRemoteIndex(t, remoteIndexRepoName, 1, 1)
+			})
+
+			it("creates an index with the underlying remote index", func() {
+				_, err = indexFactory.FetchIndex(indexRepoName, imgutil.FromBaseIndex(remoteIndexRepoName))
+				h.AssertNil(t, err)
+			})
+		})
+
+		when("index does not exist in a remote registry", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+			})
+
+			it("errors a message", func() {
+				_, err = indexFactory.FetchIndex(indexRepoName, imgutil.FromBaseIndex(indexRepoName))
+				h.AssertNotNil(t, err)
+			})
+		})
+	})
+
+	when("#FindIndex", func() {
+		when("index exists on disk", func() {
+			it.Before(func() {
+				indexRepoName = h.NewRandomIndexRepoName()
+				setUpLocalIndex(t, indexFactory, indexRepoName)
+			})
+
+			it("finds the index on disk", func() {
+				imageIndex, err = indexFactory.FindIndex(indexRepoName)
+				h.AssertNil(t, err)
+				h.AssertNotNil(t, imageIndex)
```

Comment:
Similar comment to https://github.com/buildpacks/pack/pull/1705/files#r1591429858